Feature: Add OpenTelemetry AutoConfigured SDK initialisation to TracingExtension

gustavoteixeirah opened the pull request:
The TracingExtension.java class was extended by introducing a new method to initialize an OpenTelemetrySdk instance using
AutoConfiguredOpenTelemetrySdk builder. This instance sets the global OpenTelemetry SDK, and configures various properties including tracing exporter, propagators, and disabled resource providers.

The purpose of this change is to streamline the handling of tracing in testing environment. Developers can now easily initialise the Open Telemetry SDK, which may be useful for enhanced tracking of testing processes.

Also, the pom.xml file in the common-testing directory was updated to include a dependency on 'opentelemetry-sdk-extension-autoconfigure' for enabling the AutoConfiguredOpenTelemetrySdk usage.

<!-- Example Title: "fix: [JIRA-123] Allow creation of groups with no members" -->
## Impact and Context

## Risks and Area of Effect

## Testing
- [ ] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Commit:
Feature: Add OpenTelemetry AutoConfigured SDK initialisation to TracingExtension

The TracingExtension.java class was extended by introducing a new method
to initialize an OpenTelemetrySdk instance using
AutoConfiguredOpenTelemetrySdk builder. This instance sets the global
OpenTelemetry SDK, and configures various properties including tracing
exporter, propagators, and disabled resource providers.

The purpose of this change is to streamline the handling of tracing in
testing environment. Developers can now easily initialise the Open
Telemetry SDK, which may be useful for enhanced tracking of testing
processes.

Also, the pom.xml file in the common-testing directory was updated to
include a dependency on 'opentelemetry-sdk-extension-autoconfigure' for
enabling the AutoConfiguredOpenTelemetrySdk usage.

Commit:
Merge branch 'main' of github.com:VertaAI/modeldb into teixeira/VUMM-1835

Commit:
Fix line wrapping in Javadoc for TracingExtension class

The diff changes the wrapping of lines in a paragraph from the Javadoc
for the TracingExtension class. The comment now adheres to the style
guide, which requires wrapping text so that it does not exceed a maximum
line length. This revamp improves the readability of the documentation.
Therefore, it is easier for developers to understand the functionality
of initiating an OpenTelemetrySdk instance using the
AutoConfiguredOpenTelemetrySdk builder within the TracingExtension
class.

jkwatson-verta commented on backend/common-testing/src/main/java/ai/verta/common/testing/utils/TracingExtension.java:
```diff
@@ -188,6 +191,55 @@ public void postProcessTestInstance(Object o, ExtensionContext extensionContext)
+            .build();
+    var openTelemetrySdk = autoConfiguredOpenTelemetrySdk.getOpenTelemetrySdk();
+    log.debug("openTelemetrySdk initialized = " + openTelemetrySdk);
+    return openTelemetrySdk;
```
would it make sense to set the static OpenTelemetry instance here, before returning?

gustavoteixeirah replied:
I tought I would set the static instance in the bean declaration in each test config beans file, no?

jkwatson-verta replied:
Won't all of the usages just be like:
```
var otel = TracingExtension.initializeOpenTelemetrySdk();
TracingExtension.setOpenTelemetry(otel);
```

And we could just do that in one step here, I would think.

Commit:
Set openTelemetry in TracingExtension

The method setOpenTelemetry is called after initializing
openTelemetrySdk in the TracingExtension.java. This is done to ensure
that the openTelemetrySdk is properly set up before any further
operation. It improves debugging and traceability of the program
execution.
